src/main/java/faang/school/achievement/cache/AchievementCache.java

```diff
@@ -0,0 +1,35 @@
+package faang.school.achievement.cache;
+
+import faang.school.achievement.model.Achievement;
+import faang.school.achievement.repository.AchievementRepository;
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class AchievementCache {
+    private final AchievementRepository achievementRepository;
+    private Map<String, Achievement> achievementsByName = new HashMap<>();
+
+    @PostConstruct
+    public void init() {
+        List<Achievement> achievements = achievementRepository.findAll();
+        achievements.forEach(achievement -> {
+            achievementsByName.put(achievement.getTitle(), achievement);
+        });
+    }
+
+    public Optional<Achievement> getAchievement(String title) {
+        return Optional.ofNullable(achievementsByName.get(title));
+    }
+
+    public HashMap<String, Achievement> getAchievementsByName() {
```

Comment:
Or you have 2-nd option avoid using manual map - it is the best option. Use Spring @Cachebale annotation-Spring make caching for you in repository. To use it add @EnableCaching spring boot main class. This way with this annotation you can directly use achivementRepository. like this:  @Cacheable(value = "achievements", key = "#title") for method getAchievement. For method getAllAchievment you can do the same way

Reply:
Sounds good, but I was asked to do with HashpMap . But I would rather try it, as you recommended